### ISSUE TITLE
My own learner

### ISSUE THREAD

**Issue by omiranda93** [Microsoft/CNTK#1671]
Hi, I'm trying to create my own learner by first understanding this example:
`class MySgdFast(UserLearner):

    def __init__(self, parameters, lr_schedule):
        super(MySgdFast, self).__init__(parameters, lr_schedule, as_numpy=False)

        self.new_p = {}
        self.grad_input = {}

        # we just need the batch axis
        ba = Axis.default_batch_axis()

        self.sample_count_input = input_variable((), dynamic_axes=[ba], name='count')

        lr = lr_schedule[0]  # assuming constant learning rate
        eta = lr / self.sample_count_input

        # we need one graph per parameter shape
        for param in parameters:
            p_shape = param.shape
            self.grad_input[p_shape] = input_variable(p_shape, dynamic_axes=[ba])
            self.new_p[p_shape] = param - eta * self.grad_input[p_shape]

    def update(self, gradient_values, training_sample_count, sweep_end):
        for p, g in gradient_values.items():
            new_p = self.new_p[p.shape]
            grad_input = self.grad_input[p.shape]

            data = {
                    self.sample_count_input: np.asarray(training_sample_count),
                    grad_input: g
                    }
            result = new_p.eval(data, as_numpy=False)
            shape = result.shape

            # result has the shape of a complete minibatch, but contains
            # only one tensor, which we want to write to p. This means, we
            # have to slice off the leading dynamic axes.
            static_tensor = result.data.slice_view([0]*len(shape),
                                                   shape[2:])
            p.set_value(static_tensor)

        return True`
But the thing is, first silice_view is not working. So far I've figured out that it is because shape[2:] must be changed to shape[1:], because parameters can have 1 or 2 dimensions. Making this change I´m able to reach a new error:
`RuntimeError                              Traceback (most recent call last)
/ASR/Library/external/ER/frameworks/cntk-gpu/anaconda3/envs/cntk-py35/lib/python3.5/site-packages/cntk/cntk_py.py in reset_smoothed_gradients(self)
   1953 
   1954     def reset_smoothed_gradients(self):
-> 1955         return _cntk_py.Learner_reset_smoothed_gradients(self)
   1956 
   1957     def learning_rate(self):

RuntimeError: SWIG director pure virtual method called CNTK::Learner::ResetSmoothedGradients`

How can I avoid or  solve this error?

**Comment by n17s** [comment]
Yes that example used to work until we made a breaking change. We did not catch it because it is not getting tested. 

I'm not sure why you are getting this error. 

By the way there's a much simpler way to code your own learner, using CNTK's universal_learner. See this [manual](https://github.com/Microsoft/CNTK/blob/67db47a90008a2ed86c0ac3b19e34c71cbaa4903/Manual/Manual_How_to_use_learners.ipynb) that describes how to use and write your own learners.
